scripts/build-worker.sh

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+
+# This is a generic build script for the GDK for Unity.
+# The expected usage is as follows:
+#   bash build.sh <unity_project_dir> <worker_type> <build_target> <log_file (optional)>
+
+set -e -u -x -o pipefail
+
+UNITY_PROJECT_DIR=$1
+WORKER_TYPE=$2
+BUILD_TARGET=$3
+SCRIPTING_BACKEND=$4
+LOG_FILE=${5:-} # Optional argument - will log to console otherwise
+
+# The asset cache ip cannot be hardcoded and so is stored in an environment variable on the build agent.
+# This is bash shorthand syntax for if-else predicated on the existance of the environment variable
+# where the else branch assigns an empty string.
+#   i.e. -
+#   if [ -z ${UNITY_ASSET_CACHE_IP} ]; then
+#       ASSET_CACHE_ARG="-CacheServerIPAddress ${UNITY_ASSET_CACHE_IP}"
+#   else
+#       ASSET_CACHE_ARG=""
+#   fi
+
+ASSET_CACHE_ARG=${UNITY_ASSET_CACHE_IP:+-CacheServerIPAddress "${UNITY_ASSET_CACHE_IP}"}
+
+pushd "$(dirname "$0")/../"
+    source "scripts/pinned-tools.sh"
```

Comment:
I personally prefer to have all the `source` commands at the top to have some kind of structure in the scripts (I know you probably didn't make that change ;) )